### PR TITLE
(SC-13) Define the Material-UI theme

### PR DESCRIPTION
**Pull request by RasussRogue**
**Motivation**
 
- To add a Material-UI theme for the app and unify the colors and fonts
 
**Modification**
 
- Added the Theme using ThemeProvider
- Defined the color palette of the app
- Linked the makeStyles of the various components to the Material theme
- Centered the component for a better render
 
**Remarks**
 
- The colors may be changed again in the future, mostly for ergonomic purposes
- More elements will be added to the page to make it look less empty.
- The package SVGR/Webpack was also added to usecustom SVG icons. [Link to the package](https://www.npmjs.com/package/@svgr/webpack)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"/>
+<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons"/>
 <html lang="fr">
 <head>
-    <meta charset="UTF-8" />
-    <title>Hello React!</title>
+    <meta charset="UTF-8"/>
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Rubik&display=swap" rel="stylesheet">
+    <title>Stat Check</title>
```
Comment (tkuy) [inline index.html:9]: It should be in your main page

File: src/app/App.tsx [hunk 2]
```diff
@@ -2,18 +2,51 @@ import * as React from "react";
...
-                <Route path="/champion">
-                    <ChampionPage/>
-                </Route>
-                <Route path="/">
-                </Route>
-            </Switch>
-        </Router>
+        <ThemeProvider theme={theme}>
+            <CssBaseline />
+            <Paper style={{height:'100vh', background: 'linear-gradient(180deg, rgba(2,0,36,1) 0%, rgba(30,17,89,1) 45%, rgba(30,17,89,1) 100%'}}>
```
Comment (tkuy) [inline src/app/App.tsx:38]: You have too much things in your style. makestyle ?

File: src/app/components/champion/ChampionView.tsx [hunk 3]
```diff
@@ -14,47 +24,48 @@ type ChampionProps = Readonly<{
 }>
 
 export const ChampionView: FC<ChampionProps> = ({champion, callback, championsList}) => {
-    const useStyles = makeStyles({
+    const useStyles = makeStyles((theme: Theme) => ({
         grid: {
-            fontFamily: 'Roboto'
+            fontFamily: 'Rubik',
+            color: theme.palette.common.white
```
Comment (tkuy) [inline src/app/components/champion/ChampionView.tsx:30]: It's not normal that you have to use your colors in here. It's kinda weird

File: src/app/App.tsx [hunk 4]
```diff
@@ -2,18 +2,51 @@ import * as React from "react";
...
-            <Switch>
-                <Route path="/champion">
-                    <ChampionPage/>
-                </Route>
-                <Route path="/">
-                </Route>
-            </Switch>
-        </Router>
+        <ThemeProvider theme={theme}>
+            <CssBaseline />
```
Comment (tkuy) [inline src/app/App.tsx:37]: I think you are missing a GlobalStyleProvider. It is why you have to use the colors of your theme in some component

File: src/app/App.tsx [hunk 5]
```diff
@@ -2,18 +2,51 @@ import * as React from "react";
 import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
 import {ChampionPage} from "./components/champion";
 import {Header} from "./components/commons/Header";
+import {createMuiTheme, ThemeProvider} from "@material-ui/core/styles";
+import {CssBaseline, Paper} from "@material-ui/core";
+
+const theme = createMuiTheme({
```
Comment (tkuy) [inline src/app/App.tsx:8]: Move it to an other

**Review by tkuy — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (RasussRogue) [reply index.html:9]: Change has been made, title is now defined through React.

File: src/app/App.tsx (hunk 2 above)
Comment (RasussRogue) [reply src/app/App.tsx:38]: Change has been made, added styles to a makeStyle.

File: src/app/App.tsx (hunk 4 above)
Comment (RasussRogue) [reply src/app/App.tsx]: GlobalStyleProvider has been added accordingly

File: src/app/App.tsx (hunk 5 above)
Comment (RasussRogue) [reply src/app/App.tsx]: Moved to its own file.

**Pull request merged**